src/main/java/org/truffleruby/core/numeric/BignumNodes.java

```diff
@@ -722,7 +722,7 @@ public DynamicObject pow(DynamicObject a, long b) {
                 return null; // Primitive failure
             } else {
                 // TODO CS 15-Feb-15 what about this cast?
-                return createBignum(Layouts.BIGNUM.getValue(a).pow((int) b));
+                return createBignum(pow(Layouts.BIGNUM.getValue(a),(int) b));
```

Comment:
Actually, all methods of BigInteger should be TruffleBoundary, except those that don't throw exceptions and have very few conditions. That is pretty much all methods except maybe `BigInteger.valueOf(long)`.